app/src/main/java/com/google/samples/apps/sunflower/GardenActivity.kt

```diff
@@ -52,14 +49,23 @@ class GardenActivity : AppCompatActivity() {
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         // Look for an action or destination matching the menu item ID and navigate there is found.
         // Otherwise, bubble up to the parent.
-
         return NavigationUI.onNavDestinationSelected(item,
                 Navigation.findNavController(this, R.id.garden_nav_fragment))
                 || super.onOptionsItemSelected(item)
     }
 
     override fun onSupportNavigateUp(): Boolean {
-        return NavigationUI.navigateUp(drawerLayout,
+        return NavigationUI.navigateUp(binding.drawerLayout,
                 Navigation.findNavController(this, R.id.garden_nav_fragment))
     }
+
+    override fun onBackPressed() {
```

Comment:
Since `drawerLayout` is only used twice, I feel that omitting `with` is a bit easier to read:

```
    override fun onBackPressed() {
        if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
            drawerLayout.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }
```

Reply:
ok,  I have optimized.